dist/rules/no-internal-barrel-imports.js

```diff
@@ -224,10 +229,17 @@ const rule = {
             child.moduleSpecifier !== undefined &&
             !child.isTypeOnly;
           if (!potentialReExport) return;
-          const transitiveImportInfo = ts.getResolvedModule(
-            importedModule,
-            child.moduleSpecifier.text
-          );
+          // For ts v5.3.0 and up `ts.getResolvedModule` was removed in https://github.com/microsoft/TypeScript/pull/55790
+          // Hence this check is needed to support all ts versions
+          const transitiveImportInfo = ts.getResolvedModule ?
```

Comment:
Since you are doing this twice, maybe extract to a function?

Reply:
good idea. Thanks Ben. I have extracted the function being used and the check to its own function with some helpful type directives and added to the comment explanation